src/Symfony/Component/HttpFoundation/ParameterBag.php

```diff
@@ -188,6 +188,35 @@ public function getBoolean($key, $default = false)
     }
 
     /**
+     * Returns the parameter value converted to a DateTime object.
+     *
+     * @param string             $key      The parameter key
+     * @param string             $format   The expected date format
+     * @param string             $default  The default value to be converted to a DateTime object if the parameter key does not exist
+     * @param \DateTimeZone|null $timeZone A DateTimeZone object representing the desired time zone
+     *
+     * @return \DateTime|null
+     */
+    public function getDate($key, $format, $default = null, $timeZone = null)
+    {
+        $time = $this->get($key, (string) $default);
```

Comment:
What about
```php
if (null === ($time = $this->get($key, $default))) {
    return null;
}
```

Next, if `$format` is invalid (ie `false === $result`) i would expect a `\LogicException` or something.